ha_lineairdb.cc

```diff
@@ -569,11 +812,76 @@ int ha_lineairdb::delete_all_rows() {
   the section "locking functions for mysql" in lock.cc;
   copy_data_between_tables() in sql_table.cc.
 */
-int ha_lineairdb::external_lock(THD*, int) {
+int ha_lineairdb::external_lock(THD* thd, int lock_type) {
   DBUG_TRACE;
+
+  thread = thd;
+  LineairDBTransaction*& tx = get_transaction(thd);
+  if (lock_type == F_UNLCK) {
+    if (tx->is_a_single_statement()) {
+      lineairdb_commit(lineairdb_hton, thd, true);
+    }
+    return 0;
+  }
+
+  if (tx->is_not_started()) {
+    tx->begin_transaction();
+  }
+
   return 0;
 }
 
+int ha_lineairdb::start_stmt(THD *thd, thr_lock_type lock_type) {
+  assert(lock_type > 0);
+  return external_lock(thd, lock_type);
+}
+
+/**
+ * @brief Gets transaction from MySQL allocated memory
+ */
+LineairDBTransaction*& ha_lineairdb::get_transaction(THD* thd) {
+  LineairDBTransaction *&tx = *reinterpret_cast<LineairDBTransaction**>(thd_ha_data(thd, lineairdb_hton));
+  if (tx == nullptr) {
+    tx = new LineairDBTransaction(thd, get_db(), lineairdb_hton);
+  }
+  return tx;
+}
+
+/**
+ * implementation of commit for lineairdb_hton
+*/
+static int lineairdb_commit(handlerton *hton, THD *thd, bool shouldCommit) {
+  if (shouldCommit == false) return 0;
```

Comment:
Do we need to call `cleanup_committed_tx` for the transaction when we return early here?

Reply:
No because transaction is not ended yet when it returns here.